### PR TITLE
Fix Creative Tab sorting and used better methods for adding special sub ...

### PR DESCRIPTION
**Pull request by iTitus**
...items/blocks. Will fix the NEI crash related to spawning in the NEi version of the creative KEB.

### CODE REVIEW

File: src/main/java/buildcraftAdditions/items/ItemCanister.java [hunk 1]
```diff
@@ -67,16 +71,9 @@ public void registerIcons(IIconRegister register) {
 		overlay = RenderUtils.registerIcon(register, "fluidOverlay");
 	}
 
-	@Override
-	@SideOnly(Side.CLIENT)
-	public IIcon getIconFromDamage(int damage) {
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/items/ItemCanister.java]: why is this removed?

File: src/main/java/buildcraftAdditions/items/ItemCanister.java [hunk 2]
```diff
@@ -90,4 +87,14 @@ public ItemStack getFilledItemStack(FluidStack fluidStack) {
...
 		return capacity;
 	}
+
+	@Override
+	public void getSubItems(Item item, CreativeTabs tab, List list) {
+		super.getSubItems(item, tab, list);
+		for (Fluid fluid : FluidRegistry.getRegisteredFluids().values()) {
+			if (fluid != null)
+				list.add(
+						getFilledItemStack(new FluidStack(fluid, getCapacity())));
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/items/ItemCanister.java]: fix formatting here please

File: src/main/java/buildcraftAdditions/creative/TabBCAdditions.java [hunk 3]
```diff
@@ -29,12 +27,4 @@ public Item getTabIconItem() {
 		return Item.getItemFromBlock(ItemsAndBlocks.fluidicCompressorBlock);
 	}
 
-	@Override
-	public void displayAllReleventItems(List list) {
-		super.displayAllReleventItems(list);
-		list.add(ItemsAndBlocks.powerCapsuleTier1.createdFilledBattery());
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/creative/TabBCAdditions.java]: this is needed so it shows up in the creative tab if i'm not mistaken

File: src/main/java/buildcraftAdditions/creative/TabCanisters.java [hunk 4]
```diff
@@ -42,19 +37,4 @@ public ItemStack getIconItemStack() {
 		return ItemsAndBlocks.diamondCanister.getFilledItemStack(new FluidStack(FluidRegistry.getFluid("fuel"), ItemsAndBlocks.diamondCanister.getCapacity()));
 	}
 
-	@Override
-	@SideOnly(Side.CLIENT)
-	public void displayAllReleventItems(List list) {
-		super.displayAllReleventItems(list);
-
-		createFullCanisters(ItemsAndBlocks.ironCanister, list);
```
Comment (AEnterprise) [inline src/main/java/buildcraftAdditions/creative/TabCanisters.java]: same as above

File: src/main/java/buildcraftAdditions/items/ItemCanister.java (hunk 1 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/items/ItemCanister.java]: Same as super.getIconFromDamage(int damage);

Comment (iTitus) [inline src/main/java/buildcraftAdditions/items/ItemCanister.java]: When you do a 'setTextureName()' MC will automatically load the icon and set itemIcon to this and it will properly return the icon on getIcon.

File: src/main/java/buildcraftAdditions/creative/TabCanisters.java (hunk 4 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/creative/TabCanisters.java]: Thats what getSubItems/Blocks is for :P

File: src/main/java/buildcraftAdditions/creative/TabBCAdditions.java (hunk 3 above)
Comment (iTitus) [inline src/main/java/buildcraftAdditions/creative/TabBCAdditions.java]: Thats what getSubItems/Blocks is for :P
